orc/orcarm.h

```diff
@@ -24,7 +24,23 @@ typedef enum {
   ORC_ARM_IP,
   ORC_ARM_SP,
   ORC_ARM_LR,
-  ORC_ARM_PC
+  ORC_ARM_PC,
+  ORC_ARM_R16,
+  ORC_ARM_R17,
+  ORC_ARM_R18,
+  ORC_ARM_R19,
+  ORC_ARM_R20,
+  ORC_ARM_R21,
+  ORC_ARM_R22,
+  ORC_ARM_R23,
+  ORC_ARM_R24,
+  ORC_ARM_R25,
+  ORC_ARM_R26,
+  ORC_ARM_R27,
+  ORC_ARM_R28,
+  ORC_ARM_R29,
+  ORC_ARM_R30,
+  ORC_ARM_R31
```

Comment:
Is this really ok? Isn't this required to have the correct order from reg-base?

Reply:
Yes, I think it's okay. It seems that the current ORC imp. for other architectures use the overlapped registered for both 32bit and 64bit.
For example, in orc/orcx86.h, we can see like
```
enum {
  X86_EAX = ORC_GP_REG_BASE,
  X86_ECX,
  X86_EDX,
  X86_EBX,
  X86_ESP,
  X86_EBP,
  X86_ESI,
  X86_EDI,
  X86_R8,
  X86_R9,
  X86_R10,
  X86_R11,
  X86_R12,
  X86_R13,
  X86_R14,
  X86_R15
};```
From `x86_R8`, there are x86_64 specific registers.

Reply:
![image](https://user-images.githubusercontent.com/1109011/65659031-01524b80-e065-11e9-861d-d279e6e38a91.png)

No. ```orc/orcx86.h``` has the exact ordering with register numbers from the base.

Reply:
Yes, I think x86 have confusing synonyms which are not well-matched with RXX.
But, In AAarch32/64, they have clear register names starting from R0.
Of course, they also have synonyms but according to the following figures, I think it's okay to define like suggested.
![aarch32](https://user-images.githubusercontent.com/48666767/65659869-f351fa00-e067-11e9-97dd-b5614b5cadf5.png)
![aarch64](https://user-images.githubusercontent.com/48666767/65659881-f51bbd80-e067-11e9-8352-54aa1b576d1f.png)

In the case of generic-purpose registers (i.e., R0-R30), I think it's okay. 
But, my concern is vector registers in AArch64, which are surely different from AArch32.
```
Unlike in AArch32, in AArch64 the 128-bit and 64-bit views of a SIMD and 
Floating-Point register do not overlap multiple registers in a narrower view, 
so q1, d1 and s1 all refer to the same entry in the register bank.
```